clang/lib/Lex/Preprocessor.cpp

```diff
@@ -1682,3 +1688,37 @@ const char *Preprocessor::getCheckPoint(FileID FID, const char *Start) const {
 
   return nullptr;
 }
+
+/// Whether allow C++ module directive.
+bool Preprocessor::hasSeenNoTrivialPPDirective() const {
+  return DirTracer && DirTracer->hasSeenNoTrivialPPDirective();
+}
+
+bool TrivialDirectiveTracer::hasSeenNoTrivialPPDirective() const {
+  return SeenNoTrivialPPDirective;
+}
+
+void TrivialDirectiveTracer::setSeenNoTrivialPPDirective(bool Val) {
+  if (InMainFile && !SeenNoTrivialPPDirective && Val)
```

Comment:
Ah, I see... it seems silly to have this function take an argument at all, and just not call it when the answer is 'false', since at best it does 'nothing' in that case

Reply:
Yes, exactly my thoughts.  Just not call anything in those cases.